docs/pages/blog/mui-x-v5.md

```diff
@@ -0,0 +1,191 @@
+In v5, the state is initialized synchronously during the first render.
+
+Note that for now, the state updates coming from controlled props are still asynchronous.
+If you pass `props.pageSize`, we will apply it to the state in a `useEffect`, and therefore if you read the state just after the render (for instance in a `useLayoutEffect`), you will still see the old version.
+
+## Reduced style specificity for easier customization
+
+In previous versions most of the various `DataGrid` and `DataGridPro` components had a CSS specificity of 2, meaning that style overrides and customizations were harder, requiring you to look at the DOM tree in order to pick the correct selector.
+With MUI X v5 we have reduced the CSS specificity of most of the internal `DataGrid` and `DataGridPro` components to 1.
+This will enable developers to more easily change the look and feel of the grid's components.
+
+**Before**
+
+```jsx
+const useStyles = makeStyles(() => ({
+  root: {
+    '& .MuiDataGrid-toolbarContainer': {
+      padding: 50,
+    },
+  },
+}));
+
+const MyCustomToolbar = () => {
+  return <GridToolbarContainer>My custom toolbar</GridToolbarContainer>;
+};
+
+export default function App() {
+  const classes = useStyles();
+
+  return (
+    <div style={{ height: 400, width: '100%' }} className={classes.root}>
+      <DataGrid
+        components={{
+          Toolbar: MyCustomToolbar,
+        }}
+      />
+    </div>
+  );
+}
+```
+
+**After**
+
+```jsx
+const useStyles = makeStyles(() => ({
+  root: {
+    padding: 50,
+  },
+}));
+
+const MyCustomToolbar = () => {
+  const classes = useStyles();
+
+  return (
+    <GridToolbarContainer className={classes.root}>
+      My custom toolbar
+    </GridToolbarContainer>
+  );
+};
+
+export default function App() {
+  return (
+    <div style={{ height: 400, width: '100%' }}>
+      <DataGrid
+        components={{
+          Toolbar: MyCustomToolbar,
+        }}
+      />
+    </div>
+  );
+}
+```
+
+#### Limitations
+
+Although this was a clear improvement, we still had to keep a style specificity of 2 for some parts of the `DataGrid` and `DataGridPro`, specifically the `GridColumnHeaderItem`, `GridRow` and `GridCell` along with all of the components that are nested in them.
+This is due to performance implications related to how Emotion injects styles into the page,
+and was necessary to keep the performance of our virtualization engine at its optimal.
+
+## v4 migration
+
+We strongly recommend you migrate MUI X to v5. In MUI X v5 we have not only added additional features but also made significant internal improvements and performance optimizations that won't be included in v4. All new `DataGrid` and `DataGridPro` features will be only available in MUI X v5.
+You can check [this page](https://mui.com/components/data-grid/migration-v4/) to learn more about migrating to v5.
```

Comment:
Since we are on the same domain, this could be simpler:

```suggestion
You can check [this page](/components/data-grid/migration-v4/) to learn more about migrating to v5.
```